### ISSUE TITLE
dashboard unauthorized issue with server after version 2.2.10

### ISSUE THREAD

**Issue by thuantran** [parse-community/parse-dashboard#478]
Hi there,

I'm having unauthorized issue with parse-dashboard 1.0.15 and any version of parse-server after 2.2.10. I tried all version from 2.2.11 to 2.2.17 and all of them are having unauthorized issue with my parse app. The same configuration works with parse-server 2.2.10 and lower.

I tried to create a blank mongodb database and a new parse-server instance for this db and the same issue happens.

I'm running both parse-server and parse-dashboard under pm2 with following configuration for pm2:

```
{
  "apps" : [
   {
    "name"        : "parse-dashboard",
    "script"      : "parse-dashboard",
    "watch"       : true,
    "merge_logs"  : true,
    "cwd"         : "/hosting/parse",
    "args"        : "--config dashboard-config.json"
   },
   {
    "name"        : "app-name",
    "script"      : "parse-server",
    "watch"       : true,
    "merge_logs"  : true,
    "cwd"         : "/hosting/parse",
    "env": {
      "PARSE_SERVER_CLOUD_CODE_MAIN": "/hosting/parse/cloud/main.js",
      "PARSE_SERVER_DATABASE_URI": "mongodb://user:pass@host/app-name?ssl=true",
      "PARSE_SERVER_MOUNT_PATH": "/",
      "PARSE_SERVER_MASTER_KEY": "<master key>",
      "PARSE_SERVER_APPLICATION_ID": "<application id>",
      "PARSE_SERVER_CLIENT_KEY": "<client key>",
      "PARSE_SERVER_FACEBOOK_APP_IDS": [ "app id" ]
    }
  }]
}
```

And this is the dashboard config:

```
{
  "apps": [
    {
      "allowInsecureHTTP": 1,
      "serverURL": "https://<url to parse server>",
      "appId": "<application id>",
      "masterKey": "<master key>",
      "appName": "app-name"
    }
  ],
  "users": [
    {
      "user": "user",
      "pass": "pass"
    }
  ]
}
```

I found a similar issue here https://github.com/ParsePlatform/parse-dashboard/issues/391. My configuration is returning 403 for requests on: serverInfo, _Installation, _User.

If you need further info, please do tell.

Best regards,
Thuan

**Comment by sjmn** [comment]
I'm having the same issue and spending hours spinning in circles trying to resolve it.  @thuantran have you found a solution?

**Comment by sjmn** [comment]
it seems that using the appId and masterKey for user and pass, respectively, in the dashboard config works

**Comment by peterwilli** [comment]
@sjmn thanks! I think in the last 8 hours we both were looking ... :) If it weren't for you I would be looking for a few more.

**Comment by 185driver** [comment]
I'm still having this "unauthorized" issue even after trying sjmn's temporary fix. I have multiple apps in my dashboard, so I selected one of them as my user name, and used the master key for the password as suggested. It's true that the "unauthorized" issue disappeared on the dashboard screen (and it now displays parse-server 2.2.18 as in use, but I can no longer see my data inside each app's work space, even for the app which was used as my user name login. The data display area is blank. Only when using 2.2.10 (max) does it seem to function normally.

**Comment by goggelj** [comment]
185driver: I had a similar issue with some of my classes (post migration from parse.com). On a close examination of the SCHEMA collection I noticed various 'permission attributes' differences between classes that were created during my time with parse.com vs. Classes created post migration. If your only issue is viewing data for some classes, you may want to check the SCHEMA collection for differences and make the necessary changes/adjustments.

**Comment by 185driver** [comment]
Thanks, goggelj. In this case it's the whole app work space that isn't opening.

**Comment by talkaboutdesign** [comment]
Same here.

**Comment by Sharptsa** [comment]
I had the same problem, but it's now fixed for me using both the dashboard and the server updated to the current last version.

**Comment by 185driver** [comment]
I can confirm what Sharptsa says. This issue is resolved when using Parse-server version 2.2.22 and can likely be closed.

**Comment by jpv123** [comment]
Same problem happening with last versions of both parse server and parse dashboard
